### DISCUSSION TITLE
can cubelist support add outline like cube?

### DISCUSSION THREAD

**Discussion by smbera** [foxglove/community#647] [Q&A]
this cubelist dont have outline:
![image](https://user-images.githubusercontent.com/23401125/221747806-dc9df6da-fff6-4acb-ba21-e93092d9cba2.png)

This is the expected effect I want:
![image](https://user-images.githubusercontent.com/23401125/221748014-44bd726e-39a1-4055-aa7d-9bc5917edc5b.png)

**Comment by jtbandes** [comment]
Looks like this was _almost_ implemented... https://github.com/foxglove/studio/blob/19490bb48ba5673117f045c23a2d027f1abf42c8/packages/studio-base/src/panels/ThreeDeeRender/renderables/markers/RenderableCubeList.ts#L16

**Reply by smbera** [reply]
have code implementation details？

**Reply by smbera** [reply]
https://github.com/egghunters/bim-viewer/blob/main/src/core/utils/ObjectUtils.ts#L516
I found relevant information, but it didn't work after I wrote the code.

and this is my code.
![image](https://user-images.githubusercontent.com/23401125/221752062-31f4a6c0-0917-49ea-a9ea-75ef4c8432b8.png)

**Reply by defunctzombie** [reply]
@jhurliman ^ thoughts on how we'd go about implementing this

**Reply by jtbandes** [reply]
We’ve already implemented this in some places e.g. for arrows: https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableArrows.ts

**Reply by jhurliman** [reply]
Instanced geometry is a special beast. To implement this, I believe you need to pre-generate line segments for a single cube once (maybe already done in the cube marker?) and instance it, then make this instanced geometry mirror what's going on with the cube instancing and keep them in sync.

**Reply by jtbandes** [reply]
In fact it looks like we already support outlines for CubePrimitive, just not the Marker version.

**Reply by smbera** [reply]
> In fact it looks like we already support outlines for CubePrimitive, just not the Marker version.

Thanks! I use RenderableCubes for my needs
https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableCubes.ts
